chromium_src/services/network/cookie_expiry_browsertest.cc

```diff
@@ -0,0 +1,179 @@
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    https_server_->ServeFilesFromDirectory(test_data_dir);
+    ASSERT_TRUE(https_server_->Start());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    InProcessBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+  }
+
+ protected:
+  std::unique_ptr<net::EmbeddedTestServer> https_server_;
+
+ private:
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// Set a cookie with JavaScript.
+void JSDocumentCookieWriteCookie(Browser* browser, std::string age) {
+  bool rv = content::ExecuteScript(
+      browser->tab_strip_model()->GetActiveWebContents(),
+      base::StringPrintf("document.cookie = 'name=Test; %s'", age.c_str()));
+  CHECK(rv);
+}
+
+void JSCookieStoreWriteCookie(Browser* browser, std::string expires_in_ms) {
+  content::EvalJsResult result = content::EvalJs(
+      browser->tab_strip_model()->GetActiveWebContents(),
+      base::StringPrintf("async function doSet() {"
+                         "  await window.cookieStore.set("
+                         "       { name: 'name',"
+                         "         value: 'Good',"
+                         "         expires: Date.now() + %s,"
+                         "          });"
+                         "  window.domAutomationController.send(true);"
+                         "}"
+                         "doSet()",
+                         expires_in_ms.c_str()),
+      content::EXECUTE_SCRIPT_USE_MANUAL_REPLY);
+}
+
+void SetCookieDirect(Browser* browser,
+                     const GURL& url,
+                     const std::string& cookie_line) {
+  net::CookieOptions options;
+  options.set_include_httponly();
+  options.set_same_site_cookie_context(
+      net::CookieOptions::SameSiteCookieContext::MakeInclusive());
+
+  auto cookie_obj = net::CanonicalCookie::Create(
+      url, cookie_line, base::Time::Now(), absl::nullopt /* server_time */,
+      absl::nullopt /* cookie_partition_key */);
+
+  base::RunLoop run_loop;
+  browser->tab_strip_model()
+      ->GetActiveWebContents()
+      ->GetBrowserContext()
+      ->GetDefaultStoragePartition()
+      ->GetCookieManagerForBrowserProcess()
+      ->SetCanonicalCookie(
+          *cookie_obj, url, options,
+          base::BindLambdaForTesting(
+              [&](net::CookieAccessResult status) { run_loop.Quit(); }));
+  run_loop.Run();
+}
+
+std::vector<net::CanonicalCookie> GetAllCookiesDirect(Browser* browser) {
```

Comment:
please don't pollute the global namespace. this is usually "fun" to debug when a similar function appears in another test, but with a slightly different logic.

pls either move all these function into anonymous ns or move them into a test class.